src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirModel.cs

```diff
@@ -48,238 +51,314 @@ public sealed class SqlServerFhirModel : IHostedService
-            }
+                        INSERT INTO dbo.ResourceType (Name)
+                        SELECT value FROM string_split(@resourceTypes, ',')
+                        EXCEPT SELECT Name FROM dbo.ResourceType WITH (TABLOCKX);
 
-            // result set 5
-            await reader.NextResultAsync(cancellationToken);
+                        -- result set 1
+                        SELECT ResourceTypeId, Name FROM dbo.ResourceType;
 
-            while (await reader.ReadAsync(cancellationToken))
-            {
-                var (value, systemId) = reader.ReadRow(VLatest.System.Value, VLatest.System.SystemId);
-                systemToId.TryAdd(value, systemId);
-            }
+                        INSERT INTO dbo.SearchParam (Uri)
+                        SELECT * FROM  OPENJSON (@searchParams)
+                        WITH (Uri varchar(128) '$.Uri')
+                        EXCEPT SELECT Uri FROM dbo.SearchParam;
 
-            // result set 6
-            await reader.NextResultAsync(cancellationToken);
+                        -- result set 2
+                        SELECT Uri, SearchParamId FROM dbo.SearchParam;
 
-            while (await reader.ReadAsync(cancellationToken))
-            {
-                (string value, int quantityCodeId) = reader.ReadRow(VLatest.QuantityCode.Value, VLatest.QuantityCode.QuantityCodeId);
-                quantityCodeToId.TryAdd(value, quantityCodeId);
-            }
+                        INSERT INTO dbo.ClaimType (Name)
+                        SELECT value FROM string_split(@claimTypes, ',')
+                        EXCEPT SELECT Name FROM dbo.ClaimType;
 
-            _resourceTypeToId = resourceTypeToId;
-            _resourceTypeIdToTypeName = resourceTypeIdToTypeName;
-            _searchParamUriToId = searchParamUriToId;
-            _systemToId = systemToId;
-            _quantityCodeToId = quantityCodeToId;
-            _claimNameToId = claimNameToId;
-            _compartmentTypeToId = compartmentTypeToId;
+                        -- result set 3
+                        SELECT ClaimTypeId, Name FROM dbo.ClaimType;
 
-            _started = true;
+                        INSERT INTO dbo.CompartmentType (Name)
+                        SELECT value FROM string_split(@compartmentTypes, ',')
+                        EXCEPT SELECT Name FROM dbo.CompartmentType;
+
+                        -- result set 4
+                        SELECT CompartmentTypeId, Name FROM dbo.CompartmentType;
+                        
+                        COMMIT TRANSACTION
+    
+                        -- result set 5
+                        SELECT Value, SystemId from dbo.System;
+
+                        -- result set 6
+                        SELECT Value, QuantityCodeId FROM dbo.QuantityCode";
+
+                    string searchParametersJson = JsonConvert.SerializeObject(_searchParameterDefinitionManager.AllSearchParameters.Select(p => new { Name = p.Name, Uri = p.Url }));
```

Comment:
> Name = p.Name, [](start = 149, length = 15)

What exactly the point of passing name if we extract only Uri?
```
 SELECT * FROM  OPENJSON (@searchParams)
                        WITH (Uri varchar(128) '$.Uri')
```

I know it's not your code, but if it can be cleaned, why not

Reply:
Fixed in dffb0c6.